src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,77 @@
+package com.internet.shop.web.filters;
+
+import com.internet.shop.controllers.user.LoginController;
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.interfaces.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    private Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/admin/cart/remove", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/carts/edit", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/remove", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/edit", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/edit", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/remove", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/cart/clear", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/cart/detail", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/confirm", Set.of(Role.RoleName.USER));
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String requstedUrl = req.getServletPath();
+
+        if (protectedUrls.get(requstedUrl) == null) {
```

Comment:
let's put both conditions in one `if`
also, we can use `containsKey()` here

Reply:
@Nokafan UPD: to make it work with one `if` you will have to remove `user` local variable, so this is optional

Reply:
Done. Thx. :)